homeassistant/components/mqtt/image.py

```diff
@@ -0,0 +1,246 @@
+"""Support for MQTT images."""
+from __future__ import annotations
+
+from base64 import b64decode
+import binascii
+from collections.abc import Callable
+import functools
+import logging
+import ssl
+from typing import Any
+
+import httpx
+import voluptuous as vol
+
+from homeassistant.components import image
+from homeassistant.components.image import (
+    ATTR_IMAGE_LAST_UPDATED,
+    DEFAULT_CONTENT_TYPE,
+    ImageEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, CONF_VALUE_TEMPLATE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.service_info.mqtt import ReceivePayloadType
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.util import dt as dt_util
+
+from . import subscription
+from .config import MQTT_BASE_SCHEMA
+from .const import CONF_ENCODING, CONF_QOS, CONF_TOPIC
+from .debug_info import log_messages
+from .mixins import MQTT_ENTITY_COMMON_SCHEMA, MqttEntity, async_setup_entry_helper
+from .models import MessageCallbackType, MqttValueTemplate, ReceiveMessage
+from .util import get_mqtt_data, valid_subscribe_topic
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CONTENT_TYPE = "content_type"
+CONF_IMAGE_ENCODING = "image_encoding"
+CONF_FROM_URL_TOPIC = "from_url_topic"
+
+DEFAULT_NAME = "MQTT Image"
+
+MQTT_IMAGE_ATTRIBUTES_BLOCKED = frozenset({ATTR_IMAGE_LAST_UPDATED})
```

Comment:
What's the point of this? The timestamp is used for the image entity's state, it's not exposed as a state attribute.